libs/sdk-backend-tiger/src/backend/workspaces/index.ts

```diff
@@ -53,41 +63,68 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
+                const { title } = workspace;
+
+                return title && title.toLowerCase().indexOf(lowercaseSearch) > -1;
+            });
+        }
+        return results;
+    };
+
+    private descriptorToAnalyticalWorkspace = (descriptor: IWorkspaceDescriptor): IAnalyticalWorkspace =>
+        new TigerWorkspace(this.authCall, descriptor.id, this.dateFormatter, descriptor);
+
+    private descriptorsToAnalyticalWorkspaces = (
+        descriptors: IWorkspaceDescriptor[],
+    ): IAnalyticalWorkspace[] => descriptors.map(this.descriptorToAnalyticalWorkspace);
+
     private async queryWorker(
         offset: number,
         limit: number,
         search?: string,
     ): Promise<IWorkspacesQueryResult> {
-        const emptyResult: IWorkspacesQueryResult = {
-            search,
-            items: [],
-            limit,
-            offset,
-            totalCount: 0,
-            next: () => Promise.resolve(emptyResult),
-        };
-        const workspaces = (
-            await this.authCall(async (sdk) => {
-                return sdk.organizationObjects.getAllEntitiesWorkspaces(
-                    {
-                        page: offset / limit,
-                        size: limit,
-                        sort: ["name"], // This is default behavior in gdc-bear
-                    },
-                    { headers: jsonApiHeaders },
-                );
-            })
-        ).data.data;
-        return {
-            search,
-            items: workspaces.map((workspace) => {
-                const descriptor = workspaceConverter(workspace);
-                return new TigerWorkspace(this.authCall, descriptor.id, this.dateFormatter, descriptor);
-            }),
-            limit,
-            offset,
-            totalCount: workspaces.length,
-            next: () => Promise.resolve(emptyResult),
+        const optionsToUse: OrganizationGetEntitiesOptions = {
+            query: {
+                size: limit,
+                page: offset,
+            },
         };
+
+        const allWorkspaces = await this.authCall((client) => {
+            return OrganizationUtilities.getAllPagesOf(
+                client,
+                client.organizationObjects.getAllEntitiesWorkspaces,
+                { sort: ["name"] },
+                optionsToUse,
+            )
+                .then(OrganizationUtilities.mergeEntitiesResults)
+                .then(this.resultToWorkspaceDescriptors)
+                .then(this.searchWorkspaceDescriptors(search))
+                .then(this.descriptorsToAnalyticalWorkspaces);
+        });
+
+        return new WorkspacesInMemoryPaging(allWorkspaces, limit ?? 50, 0, search);
+    }
+}
+
+class WorkspacesInMemoryPaging extends InMemoryPaging<IAnalyticalWorkspace> {
+    constructor(
+        all: IAnalyticalWorkspace[],
+        limit = 50,
+        offset = 0,
+        public readonly search: string | undefined = undefined,
+    ) {
+        super(all, limit, offset);
```

Comment:
the problem here is that InMemoryPaging creates a new instance of itself on next(). the `search` field will not be available on subsequent pages.